src/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -43,6 +45,16 @@ public static class HostWriter
             }
 
             BinaryUtils.CopyFile(appHostSourceFilePath, appHostDestinationFilePath);
+
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                var filePermissionOctal = Convert.ToInt32("755", 8); // -rwxr-xr-x
+                if (chmod(appHostDestinationFilePath, filePermissionOctal) == -1)
+                {
+                    throw new Win32Exception(Marshal.GetLastWin32Error(), $"Could not set file permission {filePermissionOctal} for {appHostDestinationFilePath}.");
+                }
+            }
```

Comment:
This code should move within the try block below (as the first thing) so that the AppHost file is deleted on failure,